src/inmanta/server/services/orchestrationservice.py

```diff
@@ -787,8 +810,8 @@ async def _put_version(
         version: int,
         rid_to_resource: dict[ResourceIdStr, data.Resource],
         unknowns: abc.Sequence[data.UnknownParameter],
+        resource_sets: dict[ResourceIdStr, data.ResourceSet],
```

Comment:
I'm not sure about this, and the resulting changes to this method. I think the whole insert and link could be done in a single database call (multiple insert queries: insert sets, insert resources, insert model). If we do it like that, we also don't need `ResourceSet` objects here, just the names.

@arnaudsjs wdyt?

Reply:
I agree.